flink-autoscaler/src/test/java/org/apache/flink/autoscaler/event/TestingEventCollector.java

```diff
@@ -42,7 +42,7 @@ public class TestingEventCollector<KEY, Context extends JobAutoScalerContext<KEY
     public final Map<String, Event<KEY, Context>> eventMap = new ConcurrentHashMap<>();
 
     @Override
-    public void handleEvent(
+    public synchronized void handleEvent(
```

Comment:
Is there a way to make this method thread-safe without synchronizing on its entirety?

Reply:
Maybe using `Collections.synchronizedList(..)`?

Reply:
Some callers use the poll methods of LinkedList, these methods from Queue, so I refactor it to `public final Queue<Event<KEY, Context>> events = new LinkedBlockingQueue<>();`.